src/Schema/Builder.php

```diff
@@ -142,13 +143,28 @@ public function dropAllTables()
         }
     }
 
-    /** @param string|null $schema Database name */
+    /** @param  string|null $schema Database name */
     public function getTables($schema = null)
     {
         $db = $this->connection->getDatabase($schema);
         $collections = [];
 
-        foreach ($db->listCollectionNames() as $collectionName) {
+        foreach ($db->listCollections() as $collectionInfo) {
+            $collectionName = $collectionInfo->getName();
+
+            // Skip system collections
+            if (str_starts_with($collectionName, 'system.')) {
+                continue;
+            }
+
+            // Skip views it doesnt suport aggregate
+            $isView = ($collectionInfo['type'] ?? '') === 'view';
+            $stats = null;
+
+            if ($isView) {
```

Comment:
The `type` property is always set, you can use [`getType`](https://github.com/mongodb/mongo-php-library/blob/9803a12f6569c86255fab1c213da21bc3408ba4d/src/Model/CollectionInfo.php#L126C21-L126C28). 

And the variable `$stats` is always set line 168, isn't it?
```suggestion
            if ($collectionInfo->getType() === 'view') {
```